kaleidoscopic-kings/backend/main.py

```diff
@@ -0,0 +1,176 @@
+import json
+import random
+import logging
+from typing import List, Dict
+from backend import path_handler
+from backend.card_format import Card, GameState, OptionOutcome
+
+logger = logging.getLogger(__name__)
+
+
+class Deck:
+    """ The deck of cards from which the user draws """
+    EVENT_CARD_TIMEOUT = 20
+    EVENT, RESPONSE, START, END_GAME = Card.CARD_TYPES
+
+    def __init__(self, all_cards: List[Card]):
+        """
+        Handles drawing cards and card responses/time-outs.
+        :param all_cards: List of all Card objects that can be used during the game.
+        """
+        self._event_cards = set()
+        self._response_cards = {}
+        self._start_card = []
+        self._game_over_card = []
+        self._build_deck(all_cards)
+        self._next_card_id = "random"
+        # Dictionary where keys are Card objects and values are ints representing current timeout
+        self._timeout_event_cards = {}
+
+    def _build_deck(self, all_cards: List[Card]):
+        for card in all_cards:
+            if card.card_type == self.EVENT:
+                self._event_cards.add(card)
+            elif card.card_type == self.RESPONSE:
+                self._response_cards[card.card_id] = card
+            elif card.card_type == self.START:
+                self._start_card.append(card)
+            elif card.card_type == self.END_GAME:
+                self._game_over_card.append(card)
+
+        self._check_valid_deck()
+
+    def _check_valid_deck(self):
+        if not self._event_cards:
+            raise Exception("No event cards found!")
```

Comment:
A more specific exception than a base `Exception` would have been nice here. I'd probably have gone with a custom exception class that subclasses an exception in the base exception hierarchy to give it a logical place. In this case, there's an issue with the value of `self._event_cards` so my first instinct would have been to look at `ValueError`.